src/fidesops/api/v1/endpoints/connection_type_endpoints.py

```diff
@@ -27,41 +29,55 @@
 logger = logging.getLogger(__name__)
 
 
-def get_connection_types(search: Optional[str] = None) -> List[str]:
+def get_connection_types(search: Optional[str] = None) -> List[ConnectionSystemTypeMap]:
     def is_match(elem: str) -> bool:
         """If a search query param was included, is it a substring of an available connector type?"""
         return search in elem if search else True
 
-    database_types: List[str] = [
-        conn_type.value
-        for conn_type in ConnectionType
-        if conn_type
-        not in [ConnectionType.saas, ConnectionType.https, ConnectionType.manual]
-        and is_match(conn_type.value)
-    ]
-    saas_types: List[str] = [
-        saas_type.value
-        for saas_type in SaaSType
-        if saas_type != SaaSType.custom and is_match(saas_type.value)
-    ]
-    connection_types: List[str] = sorted(database_types + saas_types)
+    connection_system_types: List[ConnectionSystemTypeMap] = []
+    database_types: List[str] = sorted(
+        [
+            conn_type.value
+            for conn_type in ConnectionType
+            if conn_type
+            not in [ConnectionType.saas, ConnectionType.https, ConnectionType.manual]
+            and is_match(conn_type.value)
+        ]
+    )
+    connection_system_types.extend(
+        [
+            ConnectionSystemTypeMap(identifier=item, type=SystemType.database)
+            for item in database_types
+        ]
+    )
+    saas_types: List[str] = sorted(
+        [
+            saas_type.value
+            for saas_type in SaaSType
+            if saas_type != SaaSType.custom and is_match(saas_type.value)
+        ]
+    )
+    connection_system_types.extend(
+        [
+            ConnectionSystemTypeMap(identifier=item, type=SystemType.saas)
+            for item in saas_types
+        ]
+    )
 
-    return connection_types
+    return connection_system_types
 
 
 @router.get(
     CONNECTION_TYPES,
     dependencies=[Security(verify_oauth_client, scopes=[CONNECTION_TYPE_READ])],
-    response_model=Page[str],
+    response_model=List[ConnectionSystemTypeMap],
 )
 def get_all_connection_types(
-    *, params: Params = Depends(), search: Optional[str] = None
-) -> AbstractPage[str]:
+    *, search: Optional[str] = None
```

Comment:
If we're adding the pagination `params` back into this endpoint, a query param for the type of connector could be useful. ie. `?connection_type=saas` or `?connection_type=db`.

Reply:
good idea!

Reply:
fyi we already have a `?search` param , where we filter by connection type (postgres/stripe), not system type (saas / db). 

So for this new param, I'm thinking naming it `system_type`

Reply:
I was under the impression we were just going to be giving the entire list to @chriscalhoun1974 and he was going to be handling everything else client side.

This comment is just related to client-side _sorting_, but maybe he can clarify on whether all of these pieces are needed server side.

https://github.com/ethyca/fidesops/issues/760#issuecomment-1190312249

Reply:
Correct, I will handle the sorting of the result set via client-side.

Reply:
I guess this is a question of whether we should implement convenience features vs features that are actually needed by client side now. Open to either, but will require more testing for adding the new param

Reply:
going ahead and adding it

Reply:
> I guess this is a question of whether we should implement convenience features vs features that are actually needed by client side now. Open to either, but will require more testing for adding the new param

Given these a relatively light lifts, it doesn't warrant compromising the quality of our APIs. Thanks for adding!